### PR TITLE
Trailing whitespace check

### PR DESCRIPTION
**Pull request by pineapplethief**
#### ✔ Checklist:

- [x] This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);
- [x] Tests for the changes have been added (for bug fixes / features);
- [x] Docs have been added / updated (for bug fixes / features).

### CODE REVIEW

File: src/checks/trailing_whitespace.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    fn message(&self) -> String {
+        return format!("{}: {}", self.name, self.template);
+    }
+}
+
+impl Default for TrailingWhitespaceChecker {
+    fn default() -> Self {
+        Self {
+            name: String::from("TrailingWhitespace"),
+            template: String::from("The line has trailing whitespace")
```
Comment (mstruebing) [inline src/checks/trailing_whitespace.rs:19]:
```suggestion
            template: String::from("The line has trailing whitespace"),
```

File: README.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ It checks `.env` files for problems that may cause the application to malfunctio
 &nbsp;&nbsp;&nbsp;&nbsp;✅<a href="#lowercase-key">Lowercase key</a><br />
 &nbsp;&nbsp;&nbsp;&nbsp;✅<a href="#quote-character-will-be-available-in-v200">Quote character</a> (will be available in <a href="https://github.com/dotenv-linter/dotenv-linter/issues/172">v2.0.0</a>)<br />
 &nbsp;&nbsp;&nbsp;&nbsp;✅<a href="#space-character">Space character</a><br />
+&nbsp;&nbsp;&nbsp;&nbsp;✅<a href="#trailing-whitespace">Trailing whitespace</a><br />
```
Comment (mgrachev) [inline README.md:39]:
```suggestion
&nbsp;&nbsp;&nbsp;&nbsp;✅<a href="#trailing-whitespace-will-be-available-in-v200">Trailing whitespace</a> (will be available in <a href="https://github.com/dotenv-linter/dotenv-linter/issues/172">v2.0.0</a>)<br />
```

File: README.md [hunk 3]
```diff
@@ -333,6 +334,10 @@ FOO = BAR
 FOO=BAR
 ```
 
+### Trailing whitespace
+
+Prohibits trailing whitespace at the end of the lines
```
Comment (mgrachev) [inline README.md:339]:
```suggestion
Detects if a line has a trailing whitespace.
```

File: src/checks/trailing_whitespace.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
+use crate::checks::Check;
+use crate::common::*;
+
+pub(crate) struct TrailingWhitespaceChecker {
+    template: String,
```
Comment (mgrachev) [inline src/checks/trailing_whitespace.rs:5]:
```suggestion
    template: &'a str,
```

File: src/checks/trailing_whitespace.rs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    name: String,
```
Comment (mgrachev) [inline src/checks/trailing_whitespace.rs:6]:
```suggestion
    name: &'a str,
```

File: src/checks/trailing_whitespace.rs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+impl TrailingWhitespaceChecker {
+    fn message(&self) -> String {
+        return format!("{}: {}", self.name, self.template);
+    }
+}
+
+impl Default for TrailingWhitespaceChecker {
+    fn default() -> Self {
+        Self {
+            name: String::from("TrailingWhitespace"),
```
Comment (mgrachev) [inline src/checks/trailing_whitespace.rs:18]:
```suggestion
            name: "TrailingWhitespace",
```

File: src/checks/trailing_whitespace.rs (hunk 1 above)
Comment (mgrachev) [inline src/checks/trailing_whitespace.rs:19]:
```suggestion
            template: "Trailing whitespace detected",
```

File: src/checks/trailing_whitespace.rs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    }
+}
+
+impl Check for TrailingWhitespaceChecker {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        let raw_string = &line.raw_string;
+
+        if raw_string.ends_with(' ') {
+            let warning = Warning::new(line.clone(), self.message());
+            return Some(warning);
```
Comment (mgrachev) [inline src/checks/trailing_whitespace.rs:30]:
```suggestion
            return Some(Warning::new(line.clone(), self.message()));
```

**Comment by mstruebing** [comment]
Hey @pineapplethief, any update on this regarding @mgrachev comments? :+1:

File: README.md [hunk 8]
```diff
@@ -333,6 +334,10 @@ FOO = BAR
 FOO=BAR
 ```
 
+### Trailing whitespace
```
Comment (mgrachev) [inline README.md:337]:
```suggestion
### Trailing whitespace (will be available in [v2.0.0](https://github.com/dotenv-linter/dotenv-linter/issues/172))
```

**Comment by mgrachev** [comment]
@pineapplethief Thanks a lot! ❤️

**Pull request merged**